app/workers/pubsubhubbub/distribution_worker.rb

```diff
@@ -12,10 +12,10 @@ def perform(stream_entry_id)
 
     account = stream_entry.account
     payload = AtomSerializer.render(AtomSerializer.new.feed(account, [stream_entry]))
-    domains = account.followers_domains
+    domains = account.followers_subscription_domains if stream_entry.status&.hidden?
 
     Subscription.where(account: account).active.select('id, callback_url').find_each do |subscription|
-      next unless domains.include?(Addressable::URI.parse(subscription.callback_url).host)
+      next unless !stream_entry.status&.hidden? || domains.include?(Addressable::URI.parse(subscription.callback_url).host)
```

Comment:
Yes, it's a bit unreadable, I will change that. As for WEB_DOMAIN being a subdomain of LOCAL_DOMAIN, I'm uneasy about that, that is not an assumption we have made so far, and not one GNU Social has made either. I think multiple GNU Social users have acct: URIs pointing to different domains entirely (I can think of MMN-o, but some of those can be considered aliases). It would make the check easier, though.